### PR TITLE
WIP: Windows compliance

### PR DESCRIPTION
**Pull request by 4x10m**
Related issues:
https://github.com/4x10m/coopcycle-web/pull/1
https://github.com/4x10m/coopcycle-web/pull/2
https://github.com/4x10m/coopcycle-web/pull/3

* [ ] Deploying on windows using `make install` working
* [ ] Deploy on windows using `powershell make.ps1` working
* [ ] Deploying on linux always working

For continuation of this work: see #392 

WIP
Don't merge plz

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,22 +1,22 @@
 install:
 	@printf "\e[0;32mGenerating RSA key to encrypt webtokens..\e[0m\n"
-	@mkdir -p var/jwt
+	@[ -d var/jwt ] || @mkdir -p var/jwt
```
Comment (alexsegura) [inline Makefile]: What does `[ -d var/jwt ]` do? It checks if the directory exists?

File: README.md [hunk 2]
```diff
@@ -23,16 +23,23 @@ How to run a local instance
 
     - On OSX : use [Docker for Mac](https://www.docker.com/docker-mac) which will provide you both `docker` and `docker-compose`. It doesn't rely on Virtualbox as Docker used to.
 
-    - On Windows : use [Docker for Windows](https://www.docker.com/docker-windows) which will provide you both `docker` and `docker-compose`. It doesn't rely on Virtualbox as Docker used to.
+    - On Windows : use [Docker for Windows](https://www.docker.com/docker-windows) which will provide you both `docker` and `docker-compose`. Depending on your platform, Docker could be installed as Native or you have to install Docker toolbox which use VirtualBox instead of Hyper-V causing a lot a differences in implementations. If you hace the luck to have a CPU that support native Dockern you can [share your hard disk as a virtual volume for your appliances](https://blogs.msdn.microsoft.com/stevelasker/2016/06/14/configuring-docker-for-windows-volumes/).
```
Comment (alexsegura) [inline README.md]: `s/If you hace/If you have`
`s/a CPU that support native Dockern/a CPU that supports native Docker`

File: README.md [hunk 3]
```diff
@@ -23,16 +23,23 @@ How to run a local instance
...
+
+    You will need to download openssl for generating certificates.
+    http://gnuwin32.sourceforge.net/packages/openssl.htm
+    The make script suppose that both are found [in your Path environment variable](https://www.computerhope.com/issues/ch000549.htm).
 
     - On Linux : follow [the instructions for your distribution](https://docs.docker.com/engine/installation/). `docker-compose` binary is to be installed independently. You can use Coopcycle without root privileges, to do so run `sudo usermod -aG docker your-user` (will add you to the `docker` group).
 
-* Run the install scripts.
+* Get [a Google Map API Key](https://developers.google.com/maps/documentation/javascript/get-api-key#key) and copy it. You will be asked for it when running `make install`.
```
Comment (alexsegura) [inline README.md]: Those lines are not in the `README` anymore

File: package-lock.json [hunk 4]
```diff
@@ -5261,15 +5261,6 @@
           "dev": true,
           "optional": true
         },
-        "string_decoder": {
```
Comment (alexsegura) [inline package-lock.json]: The `package-lock.json` shouldn't be part of this PR

**Comment by alexsegura** [comment]
Thank you for submitting this PR 👍

When you have finished, can you squash all your commits please?

File: Makefile (hunk 1 above)
Comment (4x10m) [reply Makefile]: Yes this is what IT does.

**Comment by 4x10m** [comment]
Not sure of what you mean but I can try. On gitlab you can automate it, not github ?!

File: package-lock.json (hunk 4 above)
Comment (4x10m) [reply package-lock.json]: Done

File: README.md (hunk 3 above)
Comment (4x10m) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (4x10m) [reply README.md]: Done

I'm testing start a review button

File: Makefile [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
-	@docker-compose run php bin/console doctrine:query:sql 'CREATE EXTENSION IF NOT EXISTS postgis_topology' --env=dev
+	@docker-compose run php php bin/console doctrine:database:create --if-not-exists --env=dev
+	@docker-compose run php php bin/console doctrine:query:sql 'CREATE EXTENSION IF NOT EXISTS postgis' --env=dev
+	@docker-compose run php php bin/console doctrine:query:sql 'CREATE EXTENSION IF NOT EXISTS postgis_topology' --env=dev
 	@printf "\e[0;32mPopulating schema..\e[0m\n"
-	@docker-compose run php bin/console doctrine:schema:create --env=dev
-	@docker-compose run php bin/demo --env=dev
-	@docker-compose run php bin/console doctrine:migrations:version --add --all
+	@docker-compose run php php bin/console doctrine:schema:create --env=dev
+	@docker-compose run php php bin/demo --env=dev
```
Comment (alexsegura) [inline Makefile]: `bin/demo` should not be executed with `php`, it is a shell script

File: Makefile (hunk 1 above)
Comment (fedegos) [reply Makefile]: Is it possible that the second "@" is wrong? I'm getting a command not found for "@mkdir". It works fine if I remove the at sign:

`@[ -d var/jwt ] || mkdir -p var/jwt`

Comment (alexsegura) [reply Makefile]: I think `@` is to make the command "silent", no?

Comment (fedegos) [reply Makefile]: Yes, but I believe it applies to the entire line. The OR is part of the command, so you don't need to silence the mkdir again. Checkout the similar command in the osrm target on line 19:
```
@[ -d var/osrm ] || mkdir -p var/osrm
```

**Comment by alexsegura** [comment]
Work continues in #508
